docs/segmentedtickerplant.md

```diff
@@ -0,0 +1,340 @@
+| STP     | Immediate     |    93k     |     89k     |
+| STP     | Memory batch  |    181k    |    174k     |
+| TorQ TP | Immediate     |    80k     |     75k     |
+| TorQ TP | Batch         |    110k    |     98k     |
+| Tick    | Immediate     |    87k     |     87k     |
+| Tick    | Batch         |    109k    |    103k     |
+
+And the following are for batched updates (note that each message contains 100 ticks):
+
+| Process | Batching Mode | Median mps | Average mps |
+| ------- | ------------- | :--------: | :---------: |
+| STP     | Default batch |    20k     |     19k     |
+| STP     | Immediate     |    19k     |     18k     |
+| STP     | Memory batch  |    21k     |     21k     |
+| TorQ TP | Immediate     |    19k     |     18k     |
+| TorQ TP | Batch         |    19k     |     18k     |
+| Tick    | Immediate     |    20k     |     17k     |
+| Tick    | Batch         |    20k     |     19k     |
+
+The first obvious thing to be noticed is that batching the updates results in greater performance as there are fewer IPC operations and disk writes, and while some insight can be gleaned from these figures the single update results provide a better comparison of the actual process code performance. The memory batching mode is the clear leader in terms of raw performance as it does not write to disk on every update. The three 'default' batching modes are roughly equivalent in terms of performance and all have similar functionality. The three Immediate modes bring up the rear in terms of raw throughput, though the STP version is the performance leader here as it stores table column names in a dictionary which can be easily accessed rather than having to read the columns of a table in the root namespace.
+
+In this set up each message contained either one or 100 ticks, each of which had 11 fields, and there was one feed pushing to one TP which had one simple subscriber and varying these figures will impact performance. Increasing the fields in each tick or the number of ticks in a message will results in more costly IPC and IO operations which will decrease performance. Having more subscribers and increasing the complexity of subscriptions, ie. having complex where clause conditions on the subscription, will also reduce performance. These are all things worth bearing in mind as one builds an application.
+
+When writing the code for these STP modes some compromise was taken between performance and maintenance. All the UPD functions are written in a standard way and have certain common elements abstracted away in namespaces, which does technically reduce performance. Also the effort made to ensure backwards compatibility means that certain left-field options were not taken advantage of, for example, sending enlisted lists from the Immediate mode rather than converting them to tables first. We will see later on how custom modes can be defined which can be more tailored to a given application.
```

Comment:
Probably remove this: 
Also the effort made to ensure backwards compatibility means that certain left-field options were not taken advantage of, for example, sending enlisted lists from the Immediate mode rather than converting them to tables first

I'm not sure that should have ever been happening (was probably a bug)